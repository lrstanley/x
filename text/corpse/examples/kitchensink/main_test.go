// Copyright (c) Liam Stanley <liam@liam.sh>. All rights reserved. Use of
// this source code is governed by the MIT license that can be found in
// the LICENSE file.

package main

import (
	"maps"
	"os"
	"slices"
	"sort"
	"testing"

	"github.com/coder/hnsw"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/lrstanley/x/text/corpse"
)

func TestMain(m *testing.M) {
	v := m.Run()
	snaps.Clean(m)
	os.Exit(v)
}

func TestCorpus(t *testing.T) {
	graph := hnsw.NewGraph[string]()
	graph.M = 25

	corp := corpse.New()
	for _, doc := range documents {
		corp.IndexDocument(doc)
	}
	snaps.MatchSnapshot(t, corp.GetTermFrequency())

	sortedKeys := slices.Collect(maps.Keys(documents))
	sort.Strings(sortedKeys)

	for _, key := range sortedKeys {
		vector := corp.CreateVector(documents[key])
		snaps.MatchSnapshot(t, vector)
		graph.Add(hnsw.MakeNode(key, vector))
	}

	cases := []struct {
		query    string
		expected []string
	}{
		{
			query:    "yellow fox",
			expected: []string{"yellow-fox", "brown-fox"},
		},
		{
			query:    "brown fox",
			expected: []string{"brown-fox", "yellow-fox"},
		},
		{
			query:    "foo bar",
			expected: []string{"foo-bar"},
		},
	}

	for _, tt := range cases {
		t.Run(tt.query, func(t *testing.T) {
			vector := corp.CreateVector(tt.query)
			snaps.MatchSnapshot(t, vector)

			results := graph.Search(vector, len(tt.expected))
			snaps.MatchSnapshot(t, results)
		})
	}
}
