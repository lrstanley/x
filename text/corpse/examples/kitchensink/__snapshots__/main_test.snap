
[TestCorpus - 1]
map[string]int{"adipiscing":1, "aliqua":1, "alphabetically":1, "amet":1, "and":1, "bar":1, "baz":1, "brown":1, "cat":1, "consectetur":1, "do":1, "dog":1, "dolor":1, "dolore":1, "eiusmod":1, "elit":1, "et":1, "fast":1, "foo":1, "fox":2, "i":1, "incididunt":1, "ipsum":1, "jumps":2, "labore":1, "lazy":1, "lorem":1, "magna":1, "on":1, "over":2, "quick":1, "sed":1, "sit":1, "slow":1, "so":1, "store":1, "tempor":1, "testing":1, "the":3, "to":1, "ut":1, "walking":1, "was":1, "working":1, "yellow":1}
---

[TestCorpus - 2]
[]float32{0, 0, 0, 0, 0, 0, 0, 0.37908196449279785, 0, 0, 0, 0.37908196449279785, 0, 0, 0, 0, 0, 0, 0, 0.27838611602783203, 0, 0, 0, 0.27838611602783203, 0, 0.37908196449279785, 0, 0, 0, 0.27838611602783203, 0.37908196449279785, 0, 0, 0, 0, 0, 0, 0, 0.4389656186103821, 0, 0, 0, 0, 0, 0}
---

[TestCorpus - 3]
[]float32{0, 0, 0, 0, 0, 0.5773502588272095, 0.5773502588272095, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.5773502588272095, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
---

[TestCorpus - 4]
[]float32{0.22941575944423676, 0.22941575944423676, 0, 0.22941575944423676, 0, 0, 0, 0, 0, 0.22941575944423676, 0.22941575944423676, 0, 0.22941575944423676, 0.22941575944423676, 0.22941575944423676, 0.22941575944423676, 0.22941575944423676, 0, 0, 0, 0, 0.22941575944423676, 0.22941575944423676, 0, 0.22941575944423676, 0, 0.22941575944423676, 0.22941575944423676, 0, 0, 0, 0.22941575944423676, 0.22941575944423676, 0, 0, 0, 0.22941575944423676, 0, 0, 0, 0.22941575944423676, 0, 0, 0, 0}
---

[TestCorpus - 5]
[]float32{0, 0, 0.29701951146125793, 0, 0.29701951146125793, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.29701951146125793, 0, 0, 0, 0, 0, 0, 0, 0.29701951146125793, 0, 0, 0, 0, 0, 0.29701951146125793, 0.29701951146125793, 0, 0.29701951146125793, 0.17196987569332123, 0.29701951146125793, 0, 0.29701951146125793, 0.29701951146125793, 0.29701951146125793, 0}
---

[TestCorpus - 6]
[]float32{0, 0, 0, 0, 0, 0, 0, 0, 0.37908199429512024, 0, 0, 0, 0, 0, 0, 0, 0, 0.37908199429512024, 0, 0.2783861458301544, 0, 0, 0, 0.2783861458301544, 0, 0, 0, 0, 0, 0.2783861458301544, 0, 0, 0, 0.37908199429512024, 0, 0, 0, 0, 0.43896564841270447, 0, 0, 0, 0, 0, 0.37908199429512024}
---

[TestCorpus/yellow_fox - 1]
[]float32{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.5919064283370972, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.806006669998169}
---

[TestCorpus/yellow_fox - 2]
[]hnsw.Node[string]{
    {
        Key:   "yellow-fox",
        Value: {0, 0, 0, 0, 0, 0, 0, 0, 0.37908199429512024, 0, 0, 0, 0, 0, 0, 0, 0, 0.37908199429512024, 0, 0.2783861458301544, 0, 0, 0, 0.2783861458301544, 0, 0, 0, 0, 0, 0.2783861458301544, 0, 0, 0, 0.37908199429512024, 0, 0, 0, 0, 0.43896564841270447, 0, 0, 0, 0, 0, 0.37908199429512024},
    },
    {
        Key:   "brown-fox",
        Value: {0, 0, 0, 0, 0, 0, 0, 0.37908196449279785, 0, 0, 0, 0.37908196449279785, 0, 0, 0, 0, 0, 0, 0, 0.27838611602783203, 0, 0, 0, 0.27838611602783203, 0, 0.37908196449279785, 0, 0, 0, 0.27838611602783203, 0.37908196449279785, 0, 0, 0, 0, 0, 0, 0, 0.4389656186103821, 0, 0, 0, 0, 0, 0},
    },
}
---

[TestCorpus/brown_fox - 1]
[]float32{0, 0, 0, 0, 0, 0, 0, 0.806006669998169, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.5919064283370972, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
---

[TestCorpus/brown_fox - 2]
[]hnsw.Node[string]{
    {
        Key:   "brown-fox",
        Value: {0, 0, 0, 0, 0, 0, 0, 0.37908196449279785, 0, 0, 0, 0.37908196449279785, 0, 0, 0, 0, 0, 0, 0, 0.27838611602783203, 0, 0, 0, 0.27838611602783203, 0, 0.37908196449279785, 0, 0, 0, 0.27838611602783203, 0.37908196449279785, 0, 0, 0, 0, 0, 0, 0, 0.4389656186103821, 0, 0, 0, 0, 0, 0},
    },
    {
        Key:   "yellow-fox",
        Value: {0, 0, 0, 0, 0, 0, 0, 0, 0.37908199429512024, 0, 0, 0, 0, 0, 0, 0, 0, 0.37908199429512024, 0, 0.2783861458301544, 0, 0, 0, 0.2783861458301544, 0, 0, 0, 0, 0, 0.2783861458301544, 0, 0, 0, 0.37908199429512024, 0, 0, 0, 0, 0.43896564841270447, 0, 0, 0, 0, 0, 0.37908199429512024},
    },
}
---

[TestCorpus/foo_bar - 1]
[]float32{0, 0, 0, 0, 0, 0.7071067690849304, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.7071067690849304, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
---

[TestCorpus/foo_bar - 2]
[]hnsw.Node[string]{
    {
        Key:   "foo-bar",
        Value: {0, 0, 0, 0, 0, 0.5773502588272095, 0.5773502588272095, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.5773502588272095, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
    },
}
---
